streampipes-extensions/streampipes-processors-transformation-jvm/src/main/java/org/apache/streampipes/processors/transformation/jvm/processor/datetime/DateTimeProcessor.java

```diff
@@ -0,0 +1,137 @@
+
+  public static final String INPUT_DATETIME_FIELD_ID = "dateTimeField";
+  public static final String OUTPUT_DATETIME_RUNTIME_NAME = "dateTime";
+  public static final String SELECTED_INPUT_TIMEZONE = "inputTimeZone";
+
+  private String[] possibleTimeZones;
+  private String streamInputDateTimeFieldName;
+  private String selectedTimeZone;
+
+  @Override
+  public DataProcessorDescription declareModel() {
+    possibleTimeZones = getTimeZoneOptions();
+    DataProcessorDescription dpd = ProcessingElementBuilder
+        .create("org.apache.streampipes.processors.transformation.jvm.timeoperator.datetime", 0)
+        .category(DataProcessorType.STRING_OPERATOR, DataProcessorType.TIME)
+        .withLocales(Locales.EN)
+        .withAssets(Assets.DOCUMENTATION, Assets.ICON)
+        .requiredStream(StreamRequirementsBuilder.create()
+            .requiredPropertyWithUnaryMapping(
+                EpRequirements.stringReq(),
+                Labels.withId(INPUT_DATETIME_FIELD_ID),
+                PropertyScope.NONE)
+            .build())
+        .requiredSingleValueSelection(Labels.withId(SELECTED_INPUT_TIMEZONE),
+            Options.from(possibleTimeZones))
+        .outputStrategy(OutputStrategies.append(
+            EpProperties.timestampProperty(OUTPUT_DATETIME_RUNTIME_NAME)))
+        .build();
+
+    return dpd;
+  }
+
+  @Override
+  public void onInvocation(ProcessorParams parameters, SpOutputCollector spOutputCollector,
+      EventProcessorRuntimeContext runtimeContext) throws SpRuntimeException {
+    ProcessingElementParameterExtractor extractor = parameters.extractor();
+    this.streamInputDateTimeFieldName = extractor.mappingPropertyValue(INPUT_DATETIME_FIELD_ID);
+    this.selectedTimeZone = extractor.selectedSingleValue(SELECTED_INPUT_TIMEZONE, String.class);
+  }
+
+  @Override
+  public void onEvent(Event event, SpOutputCollector collector) {
+    String dateTimeString = event.getFieldBySelector(streamInputDateTimeFieldName).getAsPrimitive().getAsString();
+    // DateTimeFormatter dtFormatter = detectFormat(enteredInputFormat);
+    DateTimeFormatter dtFormatter = DateTimeFormatter.ISO_DATE_TIME;
+    ZonedDateTime zdt = parseDateTime(dateTimeString, dtFormatter);
+
+    event.addField(OUTPUT_DATETIME_RUNTIME_NAME, zdt);
+    collector.collect(event);
+  }
+
+  @Override
+  public void onDetach() {
+
+  }
+
+  private ZonedDateTime parseDateTime(String dateTimeString, DateTimeFormatter dtf) {
+    ZonedDateTime zdt;
+    try {
+      zdt = ZonedDateTime.parse(dateTimeString);
+
+    } catch (DateTimeParseException e1) {
+      try {
+        LocalDateTime ldt = LocalDateTime.parse(dateTimeString, dtf);
+        ZoneId timeZoneId = ZoneId.of(selectedTimeZone);
+        zdt = ldt.atZone(timeZoneId);
+      } catch (DateTimeParseException e2) {
+        throw new RuntimeException("Could not parse DateTime String: " + dateTimeString);
+      }
+    }
+    return zdt;
+  }
+
+  private static String[] getTimeZoneOptions() {
```

Comment:
Thank you very much. I had tried to do something similar, but I could not figure out how to turn it into an Array of Strings instead of just an Array of Objects.